dotnet/src/Connectors/Connectors.OpenAI/ChatCompletion/AzureOpenAIChatCompletionService.cs

```diff
@@ -24,61 +23,60 @@ public sealed class AzureOpenAIChatCompletionService : IChatCompletionService, I
     /// <summary>
     /// Create an instance of the <see cref="AzureOpenAIChatCompletionService"/> connector with API key auth.
     /// </summary>
-    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
-    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
-    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="serviceConfig">Service configuration <see cref="OpenAIServiceConfig"/></param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     public AzureOpenAIChatCompletionService(
-        string deploymentName,
-        string modelId,
-        string endpoint,
-        string apiKey,
+        OpenAIServiceConfig serviceConfig,
         HttpClient? httpClient = null,
         ILoggerFactory? loggerFactory = null)
     {
-        this._core = new(deploymentName, endpoint, apiKey, httpClient, loggerFactory?.CreateLogger(typeof(AzureOpenAIChatCompletionService)));
+        Verify.NotNullOrWhiteSpace(serviceConfig.DeploymentName);
```

Comment:
Should we add `NotNull` check for `serviceConfig` parameter in all constructors?
```suggestion
        Verify.NotNull(serviceConfig);
        Verify.NotNullOrWhiteSpace(serviceConfig.DeploymentName);
```